### PR TITLE
Add new ReactView mixin

### PR DESCRIPTION
**Pull request by brettjonesdev**
## Feature
This PR introduces a `ReactView` mixin which can be used to create a simple Marionette ItemView which wraps one-to-one around a React component, like so: 

```javascript
const ComponentView = ReactView(
  Marionette.ItemView.extend({
    // props are the return value of serializeData()
    template(props) {
      return <HeaderComponent {...props} />
    },
  })
)
```

All the expected React and Marionette view lifecycle events are triggered at the right time. This creates a very simple way to interface between Marionette and React. 

## Other Changes
This PR also does the following:
* Enhances jest.setup.js with some of the jquery / jsdom matchers we use in `hs-app`, like `toContainText()`, etc.
* Pins Backbone and Marionette versions to the same version as `hs-app`
* Configures Jest and Storybook build config to load shim versions of Backbone and Marionette, since those libraries aren't npm-friendly (see `shims/` directory)
* Replaces the `AppWrapper` (which used Marionette 4) with `ViewWrapper`, which gives us more flexibility to create Views and pass in collection or model data from a parent
* Updates the old stories to use `Marionette.ItemView`, since `Marionette.View` is a Marionette v4 construct

## Next Steps
* The stories used for the ReactView demonstrations are very crude, could use HSDS and some more love, perhaps demonstrating an `onDeleteClick` or similar. 
* I'm a bit fuzzy on it, but I can picture us moving towards a system where we use either a vanilla `ReactView` mixin, or an `EnhancedReactView` mixin, and replace the `ReactMixin`, which has a syntax I don't particularly care for. I'll experiment with it in the coming days/weeks!

### CODE REVIEW

**Review by ItsJonQ — APPROVED** [review]
Manually tested the examples in Storybook. Checked for (re)renders. It performs as expected 👍 .

The `ReactView` component itself also makes sense to me (although, I had to familiarize myself a bit with Marionette methods).

Great job @brettjonesdev !

🚀 from me

File: src/mixins/ReactView/ReactView.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+export const COMPONENT_UPDATED = 'component:updated'
+
+const ReactView = View =>
+  View.extend({
+    initialize(...args) {
+      if (!(this instanceof Marionette.ItemView)) {
+        console.warn('ReactView mixin is intended for use with ItemViews - behavior may not match expectations')
+      }
+      View.prototype.initialize.apply(this, args)
+      this.listenTo(this, 'close', () => this.teardownComponent())
```
Comment (knicklabs) [inline src/mixins/ReactView/ReactView.js]: Is there a reason we have to use an anonymous function here in the third position? I think we can just use `this.teardownComponent`.

File: src/mixins/ReactView/ReactView.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    _triggerBeforeRendered() {
+      this.triggerMethod('before:render', this)
+      this.triggerMethod('item:before:render', this)
+    },
+
+    _triggerRendered() {
+      this.triggerMethod('render', this)
+      this.triggerMethod('item:rendered', this)
+    },
```
Comment (knicklabs) [inline src/mixins/ReactView/ReactView.js:37]: I take it these are meant to be private methods. What do you think about moving them outside the view? Can define them like: 

```
const triggerRendered(view) {
  view.triggerMethod('render', view)
  view.triggerMethod('item:rendered', view)
}
```

Comment (brettjonesdev) [reply src/mixins/ReactView/ReactView.js:37]: Basically, yeah. I really kind of hate this way to make "private methods" lol. I get _why_, but passing the `view` instance in just kills me. I really don't care very much about protecting "private" methods in JS, I think I'd rather just make them `this.triggerRendered()` than do the passing-this-in-as-a-param thing. I just really hate that way of doing private methods in JS, idk what do you think?

**Review by knicklabs — CHANGES_REQUESTED** [review]
This is awesome @brettjonesdev! Love that you included stories for it to serve as some nice code examples. 

I had a couple tiny questions. Aside from that, would you mind adding a doc (doesn't have to be extensive), explaining what ReactView with some guidance around when to use it? 

As part of the 1.0 release, I can clean up the docs so that the Mixin and ReactView live side by side, but a one-pager like that would be a great start. 

Edit: Here's an example of a doc https://github.com/helpscout/brigade/blob/1.0/docs/03-components.mdx

File: src/mixins/ReactView/ReactView.js (hunk 2 above)
Comment (knicklabs) [reply src/mixins/ReactView/ReactView.js:37]: Was just a thought. I'm not too fussed about this as these methods seem pretty harmless and won't be documented. Fine with keeping it as is.

File: src/mixins/ReactView/ReactView.js (hunk 1 above)
Comment (brettjonesdev) [reply src/mixins/ReactView/ReactView.js]: Yep and done!

**Review by knicklabs — APPROVED** [review]

**Pull request merged**